url75694980d5_fabioalmeid_tp01ftc_tgz-pJ8-agentes_jade_Centralizador_ExemploParserGramaticaCentralizadorJ8/src/gramatica/Atuador/PrettyPrinter.java

```diff
@@ -0,0 +1,241 @@
+  {
+    sh(foo);
+    String temp = buf_.toString();
+    buf_.delete(0,buf_.length());
+    return temp;
+  }
+  public static String print(gramatica.Atuador.Absyn.Acao foo)
+  {
+    pp(foo, 0);
+    trim();
+    String temp = buf_.toString();
+    buf_.delete(0,buf_.length());
+    return temp;
+  }
+  public static String show(gramatica.Atuador.Absyn.Acao foo)
+  {
+    sh(foo);
+    String temp = buf_.toString();
+    buf_.delete(0,buf_.length());
+    return temp;
+  }
+  public static String print(gramatica.Atuador.Absyn.Remedio foo)
+  {
+    pp(foo, 0);
+    trim();
+    String temp = buf_.toString();
+    buf_.delete(0,buf_.length());
+    return temp;
+  }
+  public static String show(gramatica.Atuador.Absyn.Remedio foo)
+  {
+    sh(foo);
+    String temp = buf_.toString();
+    buf_.delete(0,buf_.length());
+    return temp;
+  }
+  /***   You shouldn't need to change anything beyond this point.   ***/
+
+  private static void pp(gramatica.Atuador.Absyn.Tarefa foo, int _i_)
+  {
+    if (foo instanceof gramatica.Atuador.Absyn.ETarefa)
+    {
+       gramatica.Atuador.Absyn.ETarefa _etarefa = (gramatica.Atuador.Absyn.ETarefa) foo;
+       if (_i_ > 0) render(_L_PAREN);
+       pp(_etarefa.acao_, 0);
+       if (_i_ > 0) render(_R_PAREN);
+    }
+  }
+
+  private static void pp(gramatica.Atuador.Absyn.Acao foo, int _i_)
+  {
+    if (foo instanceof gramatica.Atuador.Absyn.EAcao)
+    {
+       gramatica.Atuador.Absyn.EAcao _eacao = (gramatica.Atuador.Absyn.EAcao) foo;
+       if (_i_ > 0) render(_L_PAREN);
+       render("Liberando");
+       pp(_eacao.remedio_, 0);
+       if (_i_ > 0) render(_R_PAREN);
+    }
+    else     if (foo instanceof gramatica.Atuador.Absyn.EAcao1)
+    {
+       gramatica.Atuador.Absyn.EAcao1 _eacao1 = (gramatica.Atuador.Absyn.EAcao1) foo;
+       if (_i_ > 0) render(_L_PAREN);
+       render("Cessando Liberacao de");
+       pp(_eacao1.remedio_, 0);
+       if (_i_ > 0) render(_R_PAREN);
+    }
+  }
+
+  private static void pp(gramatica.Atuador.Absyn.Remedio foo, int _i_)
+  {
+    if (foo instanceof gramatica.Atuador.Absyn.ERemedio)
+    {
+       gramatica.Atuador.Absyn.ERemedio _eremedio = (gramatica.Atuador.Absyn.ERemedio) foo;
+       if (_i_ > 0) render(_L_PAREN);
+       render("Dipirona");
+       if (_i_ > 0) render(_R_PAREN);
+    }
+    else     if (foo instanceof gramatica.Atuador.Absyn.ERemedio1)
+    {
+       gramatica.Atuador.Absyn.ERemedio1 _eremedio1 = (gramatica.Atuador.Absyn.ERemedio1) foo;
+       if (_i_ > 0) render(_L_PAREN);
+       render("Paracetamol");
+       if (_i_ > 0) render(_R_PAREN);
+    }
+  }
+
+
+  private static void sh(gramatica.Atuador.Absyn.Tarefa foo)
+  {
+    if (foo instanceof gramatica.Atuador.Absyn.ETarefa)
+    {
+       gramatica.Atuador.Absyn.ETarefa _etarefa = (gramatica.Atuador.Absyn.ETarefa) foo;
+       render("(");
+       render("ETarefa");
+       sh(_etarefa.acao_);
+       render(")");
+    }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 163:169, 175:181, 183:189.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.